src/back/routes/user.ts

```diff
@@ -89,3 +91,27 @@ async function isValidated(req: Request) {
     throw new Error(`${message}. ${error}`)
   }
 }
+
+async function getProfile(req: Request) {
+  const address = req.params.address
+  validateAddress(address)
+
+  try {
+    const user = await UserModel.findOne<UserAttributes>({ address: address.toLowerCase() })
+
+    if (!user) {
+      throw new RequestError('User not found', RequestError.NotFound)
+    }
+
+    const { forum_id } = user
+
+    return {
+      forum_id,
+      forum_username: forum_id ? (await DiscourseService.getUserById(forum_id))?.username : null,
+    }
+  } catch (error) {
+    const message = 'Error while fetching profile data'
+    ErrorService.report(message, { error })
+    throw new Error(`${message}. ${error}`)
```

Comment:
Might be good to include the forum_id in the error report

Reply:
Added address here.